python/ray/util/client/common.py

```diff
@@ -60,7 +62,51 @@ def __del__(self):
 
 
 class ClientObjectRef(ClientBaseRef):
-    pass
+    def __await__(self):
+        return self.as_future().__await__()
+
+    def as_future(self) -> asyncio.Future:
```

Comment:
can we adopt the new `ref.future() -> concurrent.futures.Future` API? see object_ref.pyx

Reply:
Yep, ptal

Reply:
yeah lg!